pkg/gadgets/snapshot/socket/tracer/tracer.go

```diff
@@ -35,6 +35,24 @@ import (
 //go:generate go run github.com/cilium/ebpf/cmd/bpf2go -target bpfel -cc clang iterTCPv4 ./bpf/tcp4-collector.c -- -I../../../../${TARGET} -Werror -O2 -g -c -x c
 //go:generate go run github.com/cilium/ebpf/cmd/bpf2go -target bpfel -cc clang iterUDPv4 ./bpf/udp4-collector.c -- -I../../../../${TARGET} -Werror -O2 -g -c -x c
 
+type Tracer struct {
+	iters map[socketcollectortypes.Proto]*link.Iter
+
+	// visitedNamespaces is a map where the key is the netns inode number and
+	// the value is the pid of one of the containers that share that netns. Such
+	// pid is used by NetnsEnter. TODO: Improve NetnsEnter to also work with the
+	// netns directly.
+	visitedNamespaces map[uint64]uint32
+	protocols         socketcollectortypes.Proto
+	eventHandler      func([]*socketcollectortypes.Event)
+}
+
+func (t *Tracer) Stop() {
+	for _, it := range t.iters {
+		it.Close()
```

Comment:
#1408 moves from `gadgets.StartStopGadget` to just have a blocking `Run()` in all gadgets and optionally `Init()` and `Close()` (see also [here](https://github.com/inspektor-gadget/inspektor-gadget/pull/1408/files#diff-cf9f0e218778da1c45350fbdb0053479f23d66a86800100810a98a44d0fb8dd2)). So I think it would be better to move the `Stop()` code into `Run()`.

Reply:
Ok, I moved the code in Run() as much as possible. I still have a `CloseIters()` exported temporarily for kubectl-gadget but we can remove that as soon as #1408 is completed.